Extend `langchain.callbacks` documentation

alvarobartt opened the pull request:
Hi to whoever is reading this! 🤗

# What's in this PR?

As of the recent addition of `ArgillaCallbackHandler` in the `master` branch of `LangChain`, merged by @dev2049 (but maybe also relevant to @agola11 since it's callback-related), @davidberenstein1957 and @dvsrepo noticed that the `Callbacks -> How to guides -> Argilla` page was missing, so I've extended the `langchain.callbacks` documentation a little bit to include the "How to guides" to also allow any other callback integration to live there, and add the missing API Reference for the callbacks.

alvarobartt committed:
Add `langchain.callbacks` API reference

alvarobartt committed:
Create `callbacks/how_to_guides.rst`

alvarobartt committed:
Add `modules/callbacks.rst`

agola11 reviewed (approved):
Thanks!

leo-gan commented on docs/modules/callbacks.rst:
```diff
@@ -0,0 +1,27 @@
+=========
+
+.. note::
+   `Conceptual Guide <https://docs.langchain.com/docs/components/callbacks>`_
```
This (and others, below) link should be relative, like `./callbacks/how_to_guides.html`, no?

alvarobartt replied:
Same as in the other comment you left, did it this way because that's what I saw, but makes sense to replace that with the relative path to the file itself rather than the URL 👍🏻

alvarobartt replied:
I checked and the conceptual guide is automatically generated so there's no actual file to point to, we can only point to the URL itself, see https://raw.githubusercontent.com/hwchase17/langchain/master/docs/modules/agents/agents.rst or any other file under `docs/modules/` for reference 👍🏻

leo-gan commented on docs/modules/callbacks/how_to_guides.rst:
```diff
@@ -0,0 +1,28 @@
+=============
+
+LangChain provides a callbacks system that allows you to hook into the various stages of your LLM application. 
+This is useful for logging, `monitoring <https://python.langchain.com/en/latest/tracing.html>`_, 
```
The same. This (and others, below) link should be relative, like ./callbacks/how_to_guides.html, no?

alvarobartt replied:
Makes a lot of sense! I just used those as I saw it on another `how_to_guides.rst` file, but maybe it's worth double-checking and replacing those URLs if needed 👍🏻

alvarobartt committed:
Fix path to `ArgillaCallbackHandler` example

Co-authored-by: Daniel Vila Suero <daniel@argilla.io>

alvarobartt committed:
Fix link to Callbacks in `index.rst`

alvarobartt committed:
Fix to use relative paths instead of URL

Co-authored-by: Leonid Ganeline <leo.gan.57@gmail.com>

alvarobartt commented:
Hi again @leo-gan, I've fixed the paths to point to the relative path instead of the URL to be generated, but I saw some other references in other pages (not edited in this PR) also point to the URLs to be generated instead of the relative path, is that something we should look into?

alvarobartt committed:
Fix paths/URLs to `tracing`

alvarobartt committed:
Fix path to `Getting Started`

alvarobartt committed:
Rename notebook title to `Getting Started`

alvarobartt committed:
Revert to use URL to`Conceptual Guide`

All the modules use the URL, as it's not any existing HTML/Markdown/reStructuredText file to point to, but an auto-generated URL

alvarobartt committed:
Align `callbacks/getting_started.ipynb` with the rest

All the Jupyter Notebooks under `docs/modules` have no direct URLs to LangChain, and the introduction was duplicated, so it has been removed

alvarobartt committed:
Add missing "Advanced" note in `callbacks/how_to_guides.rst`

alvarobartt committed:
Update `callbacks/getting_started.ipynb` intro

alvarobartt commented:
Hi @leo-gan and @agola11 I've already addressed the comments, and deployed the docs locally and those now work fine! Feel free to merge whenever you want 😄

leo-gan commented on docs/modules/callbacks.rst:
```diff
@@ -0,0 +1,26 @@
+=========
+
+.. note::
+   `Conceptual Guide <https://docs.langchain.com/docs/components/callbacks>`_
```
<https://docs.langchain.com/docs/components/callbacks>
Can you make it a relative link? Thanks

alvarobartt replied:
Hi @leo-gan, as of the comment above, it doesn't work with a relative link since it's automatically generated on docs build-time, so there's nowhere to point to AFAIK, check other similar files e.g. https://github.com/hwchase17/langchain/blob/master/docs/modules/agents.rst

alvarobartt replied:
But I indeed changed the rest of the URLs to be relative paths to the actual files instead, as you suggested in your comments above 🤗

leo-gan replied:
Yeah. It points to a different site. My bad :)

leo-gan replied:
LGTM

leo-gan reviewed (approved):
LGTM

alvarobartt committed:
Merge branch 'master' into add-callbacks-docs

alvarobartt commented:
Cool, then I'm already done, feel free to merge whenever applicable! 🤗

alvarobartt committed:
Merge remote-tracking branch 'upstream/master' into add-callbacks-docs

alvarobartt commented:
As of https://github.com/hwchase17/langchain/pull/6300, the docs skeleton has changed and those issues have already been solved, so this PR will be closed! 😄 Thanks for the support in the meantime @leo-gan and @agola11
